net-utils/src/ip_echo_client.rs

```diff
@@ -106,39 +107,38 @@ fn parse_response(
 
 pub(crate) const DEFAULT_RETRY_COUNT: usize = 5;
 
-/// Checks if all of the provided TCP/UDP ports are not reachable by the machine at
+/// Checks if all of the provided TCP ports are reachable by the machine at
 /// `ip_echo_server_addr`. Tests must complete within timeout provided.
-/// Tests will run in parallel when possible.
-pub(crate) async fn do_verify_reachable_ports(
+/// Tests will run concurrently to avoid head-of-line blocking. Will return false on any error.
+pub(crate) async fn verify_reachable_tcp_ports(
     ip_echo_server_addr: SocketAddr,
-    tcp_listeners: Vec<(u16, TcpListener)>,
-    udp_sockets: &[&UdpSocket],
+    listeners: Vec<(u16, TcpListener)>,
     timeout: Duration,
-    udp_retry_count: usize,
 ) -> bool {
-    info!(
-        "Checking that tcp ports {:?} are reachable from {:?}",
-        tcp_listeners, ip_echo_server_addr
-    );
+    let mut checkers = Vec::new();
+    let mut ok = true;
 
-    let tcp_ports: Vec<_> = tcp_listeners.iter().map(|(port, _)| *port).collect();
-    let _ = ip_echo_server_request(
-        ip_echo_server_addr,
-        IpEchoServerMessage::new(&tcp_ports, &[]),
-    )
-    .await
-    .map_err(|err| warn!("ip_echo_server request failed: {}", err));
+    // Chunk the port range into slices small enough to fit into one packet
+    for chunk in &listeners.into_iter().chunks(MAX_PORT_COUNT_PER_MESSAGE) {
+        let (ports, listeners): (Vec<_>, Vec<_>) = chunk.unzip();
+        info!(
+            "Checking that tcp ports {:?} are reachable from {:?}",
+            &ports, ip_echo_server_addr
+        );
 
-    let mut ok = true;
-    let mut checkers = Vec::new();
+        // make request to the echo server
+        let _ = ip_echo_server_request(ip_echo_server_addr, IpEchoServerMessage::new(&ports, &[]))
```

Comment:
it is not directly related to this comment, but a side note. I think that this IpEchoServerMessage should not take both UDP and TCP ports at once. If we still need UPD part, it should be under a separate structure.

Reply:
Yes, but changing this would break the world. A lot of validators use this to bootsrtap.